src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -492,6 +496,10 @@ impl CloudHypervisorInner {
                 return Err(anyhow!(GuestProtectionError::NoProtectionAvailable));
             } else if let GuestProtection::Tdx(_) = protection {
                 info!(sl!(), "guest protection available and requested"; "guest-protection" => protection.to_string());
+            } else {
+                return Err(anyhow!(GuestProtectionError::ExpectedTDXProtection(
+                    protection
+                )));
```

Comment:
Is the check on line 506 correct? Isnt the condition checking for TDX and then erroring out about TDX not being used. Let me know if I misunderstood. 
Also, I feel that when confidential_guest is not requested, we should not really error out on the protection value. We could just log the protection value. In fact, in case confidential_guest is not requested, it is really not required to compute the protection value at all. What do you think?

Also, do you think you can squash the first and second commit to make this easier to review?

Reply:
Hi @amshinde - Do you mean line 556? This part of the diff does look a little bit odd as a result of inverting the logic (at your request ;)

Here's a summary of the conditions the code and unit tests handle:

| `confidential_guest` | available protection | result | notes |
|-|-|-|-|
| `false` | none | Ok | Log a message that no protection available |
| `false` | TDX | Error | CH requires TDX protection be used if available |
| `false` | non-TDX | Ok | Log available but unused protection |
| `true` | none | Error | The user requested protection, but none available, so we must fail |
| `true` | TDX | Ok | Log a message that protection available and requested |
| `true` | non-TDX | Error | The CH driver is only currently tested with TDX |

Reply:
@amshinde - I've now combined the first two commits as requested :smile: